modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFolderItemLocalServiceImpl.java

```diff
@@ -151,7 +158,37 @@ public List<ObjectFolderItem> getObjectFolderItemsByObjectDefinitionId(
 	public List<ObjectFolderItem> getObjectFolderItemsByObjectFolderId(
 		long objectFolderId) {
 
-		return objectFolderItemPersistence.findByObjectFolderId(objectFolderId);
+		DSLQuery dslQuery = DSLQueryFactoryUtil.select(
+			ObjectFolderItemTable.INSTANCE
+		).from(
+			ObjectFolderItemTable.INSTANCE
+		).where(
+			() -> {
+				Predicate predicate =
+					ObjectFolderItemTable.INSTANCE.objectFolderId.eq(
+						objectFolderId);
+
+				PermissionChecker permissionChecker =
+					PermissionThreadLocal.getPermissionChecker();
+
+				if (permissionChecker == null) {
+					return predicate;
+				}
+
+				Predicate permissionPredicate =
+					_inlineSQLHelper.getPermissionWherePredicate(
```

Comment:
Hey @pedro-oliveira446 , what happens if _SQL Check Enabled_ is turned off? Maybe you should include some piece of code like [this one](https://github.com/liferay/liferay-portal/blob/2b6e533dba0b42c5c215935188f03065a3ce744e/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java#L2801-L2806).
You can configure this on "**System Settings > Security Tools > SQL Check Enabled**"

Reply:
Another thing, should we pass the `groupId` when calling `_inlineSQLHelper.getPermissionWherePredicate`? 
You can ask Gabriel if this makes any difference in this context.